applications/luci-app-throughput-measurements/htdocs/luci-static/resources/view/tputmeas/server.js

```diff
@@ -0,0 +1,180 @@
+			if (textOut)
+				textOut.style.borderColor = "red";
+			var startBtn = document.querySelectorAll('[id$="start_iperf3"]')[0];
+			if (startBtn)
+				startBtn.hidden = false;
+			var stopBtn = document.querySelectorAll('[id$="stop_iperf3"]')[0];
+			if (stopBtn)
+				stopBtn.hidden = true;
+			if (eventSource)
+				eventSource.close();
+	}));
+
+	return  Promise.all(tasks);
+}
+
+return L.view.extend({
+
+	load: function() {},
+
+	render: function() {
+		var m, s, o;
+
+		m = new form.Map('tputmeas', _('Throughput Measurements - Iperf3 Server'), _('Iperf3 is a tool for performing network throughput measurements. It can test TCP, UDP, or SCTP throughput. To perform an iperf3 test the user must establish both a server and a client.'));
+		s = m.section(form.TypedSection, 'iperf3')
+		s.anonymous = 1;
+
+		o = s.option(widgets.DeviceSelect, 'interface', _('Interface'), _('Bind to the specific interface associated with address host.'))
+		o.noaliases = true;
+
+		o = s.option(form.Value, 'interface_address', _('Interface address'), _('Bind to the specific interface address associated with address host.'));
+		o.optional = false;
+		o.datatype = 'ip4addr("nomask")';
+		o.depends('interface', undefined);
+
+		o = s.option(form.Value, 'port', _('Port'), _('Set server port to listen on/connect to to n (default 5201)'));
+		o.modalonly = false;
+		o.datatype = 'list(neg(port))';
+
+		o = s.option(form.Button, 'start_iperf3', _('Start iperf3'), _(''));
+		o.inputstyle = 'apply';
+		o.onclick = ui.createHandlerFn(this, function(section_id, ev) {
+			var iface = document.querySelectorAll('[id$="interface"]')[1].value,
+				interface_address = document.querySelectorAll('[id$="interface_address"]')[2].value,
+				port = document.querySelectorAll('[id$="port"]')[2].value;
+
+			return callIperf3Start("-s", "", "", interface_address, port, 
+				"", "", "", "", "", "", "").then(function(replay) {
+				var error_position = replay.result.search("error:");
+				if (error_position != -1){
+					ui.showModal(_(replay.result.substring(error_position + 6, replay.result.length)), [
+						E('div', { 'class': 'right' }, [
+							E('button', {
+								'class': 'cbi-button cbi-button-negative important',
+								'click': function(ev) {
+									ui.hideModal();
+								}
+							}, _('Close')),
+						])
+					]);
+					return;
+				}
+				rpc.list.apply(rpc).then(function(res) {
+					for (var k in res) {
+						if (res[k] == "luci.tputmeas.notify")
+							subscribeIperf3()
+					}
+				}.bind(this));
+			}.bind(this)).catch(function(error) {
+				console.log(error);
+			});
+		});
+
+		o = s.option(form.Button, 'stop_iperf3', _('Stop iperf3'), _(''));
+		o.inputstyle = 'apply';
+		o.onclick =  ui.createHandlerFn(this, function(section_id, ev) {
+			if (!eventSource)
+				return;
+			return callIperf3Stop().then(function(replay) {
+				eventSource.close();
+			}.bind(this)).catch(function(error) {
+				console.log(error);
```

Comment:
Do we need this log

Reply:
No, exception will be visible in console anyway